### DISCUSSION TITLE
Integration of dnsmasq-regex patch (or a corresponding package alternative)

### DISCUSSION THREAD

**Discussion by Ni-ky** [Freetz-NG/freetz-ng#442] [Unerfüllte Wünsche]
Looking into alternatives to quite large and _.hosts_ file for ad blocking (_dnsmasq_ + _addhole_ combo), I stumbled upon [_dnsmasq-regex_](https://github.com/lixingcong/dnsmasq-regex) patch. I believe it would help users to create a more efficient and effective ad-blocking solutions based on regular expressions, comparable to that of _Pi-hole_ (less the GUI, of course).

I am lacking competencies to integrate the patches and/or packages into Freetz-NG on the quality level remotely compatible with the rest of the project. Thus I was wondering if there are fellow members of the Freetz-NG community interested to have dnsmasq@Freetz extended by the regex capability willing to contribute.

**Comment by fda77** [comment]
It should be enough if you put the files from https://github.com/lixingcong/dnsmasq-regex/tree/master/patches to make/dnsmasq/patches and recompile, if they fit and the dnsmasq versions are not to different.
Adding huge external patches make a version bump of later dnsmasq versions hard, as it could be the patches dont fit anymore and needs to be fixed first, which could be easy or not. This could block dnsmasq updates for a while.
Additionaly "ipset" seems to be a part of iptables which does not work well with avm's Fritzos, you could not replace kernel for all devices. Im even not sure if the ipset command is available in freetz.
So someone interested in it should maintain the patches, but i think theres nobody
The best would be to send the regex patches to dnsmasq to get integrated there! Then with new versions of dnsmasq they could be used in freetz with
